org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -315,7 +315,7 @@ private void deleteInvalidProjects(Collection<IPath> rootPaths, IProgressMonitor
 	private boolean isReferencedByGradleProjects(IProject target, List<IProject> gradleProjects) {
```

Comment:
You could the buildship's persistent model as follows :
```
PersistentModel model = CorePlugin.modelPersistence().loadModel(project);
if (model.isPresent()) {
  Collection<IPath> paths = model.getSubprojectPaths();
 ... 
}
```

Reply:
Thanks for the suggestion. I have changed to use the persist model to check the submodule relatioship. 
@snjeza  Could you take a look again?